.github/workflows/centos7.yml

```diff
@@ -43,7 +43,28 @@ jobs:
         yum install -y centos-release-scl
         yum install -y devtoolset-10-gcc*
 
-    - uses: ./.github/workflows/install-cmake-328
+    - name: Install cmake 3.28
+      run: pip3 install cmake==3.28.4
+
+    - name: Install VCPKG
+      # Note: we need to use environment variables instead of workflow variables
+      #       because github messes up path variables when running in container,
+      #       see https://github.com/actions/runner/issues/2058
+      run: |
+        git submodule update --init vcpkg && ./vcpkg/bootstrap-vcpkg.sh -disableMetrics
+        echo "VCPKG_ROOT=$GITHUB_WORKSPACE/vcpkg" >> $GITHUB_ENV
+        echo "VCPKG_CACHE_DIR=$GITHUB_WORKSPACE/vcpkg_cache" >> $GITHUB_ENV
+        echo "VCPKG_BINARY_SOURCES=clear;files,$GITHUB_WORKSPACE/vcpkg_cache,readwrite" >> $GITHUB_ENV
+
+    - name: Restore vcpkg binary dir from cache
+      id: cache-vcpkg-binary
+      # Note: we are stuck with v3, because v4 is not compatible with oracle8 image
+      uses: actions/cache/restore@v3
+      with:
+        path: ${{ env.VCPKG_CACHE_DIR }}
+        key: vcpkg-cache-centos7-${{ hashFiles('src/vcpkg.json', '.git/modules/vcpkg/HEAD') }}
+        # Allows to restore a cache when deps have only partially changed (like adding a dependency)
+        restore-keys: vcpkg-cache-centos7-
```

Comment:
wow! i will recommend to place the "install vcpkg and the restore cache" jobs in dedicated file